### ISSUE TITLE
Split APKs by architecture/screensize/language etc.

### ISSUE THREAD

**Issue by aegis123** [flutter/flutter#20604]
Is it possible to use the APK split or app bundle functionality in the Android Gradle plugin to reduce the size of an release APK like have splits for different cpu architectures arm64, x86 or even more advanced splits on screensize and language.

**Comment by aegis123** [comment]
@goderbauer created a new ticket

**Comment by apaatsio** [comment]
Adding `splits` config to `android/app/build.gradle` seems to produce the correct APKs in `build/app/outputs/apk/release/`. `flutter build apk` command fails but only after it has built the APKs.

1. Add splits config to `android/app/build.gradle`:

```groovy
android {
    // ...
    buildTypes {
        release {
            // ...
            splits {
                abi {
                    enable true
                    reset()
                    include "armeabi-v7a", "arm64-v8a"
                    universalApk true
                }
            }
        }
    }
}
```
2. Run `flutter build apk`. Ignore the build error.
3. Observe `build/app/outputs/apk/release/`. There should be three files:
- `app-arm64-v8a-release.apk`
- `app-armeabi-v7a-release.apk`
- `app-universal-release.apk`
---
The error I get in step 2 is
```
Gradle build failed to produce an Android package.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:24:3)
#1      _buildGradleProjectV2 (package:flutter_tools/src/android/gradle.dart:484:7)
<asynchronous suspension>
#2      buildGradleProject (package:flutter_tools/src/android/gradle.dart:346:14)
<asynchronous suspension>
#3      buildApk (package:flutter_tools/src/android/apk.dart:34:9)
<asynchronous suspension>
#4      BuildApkCommand.runCommand (package:flutter_tools/src/commands/build_apk.dart:45:11)
<asynchronous suspension>
...
```
---
```
[✓] Flutter (Channel dev, v1.4.7, on Linux, locale en_US.UTF-8)
    • Flutter version 1.4.7 at /home/antti/tools/flutter
    • Framework revision 1bfa2f2311 (3 days ago), 2019-03-29 10:05:18 -0700
    • Engine revision c4d14a0071
    • Dart version 2.2.1 (build 2.2.1-dev.2.0 None)
```

**Comment by aegis123** [comment]
How are the size compared to the universal version of the apk?

**Comment by apaatsio** [comment]
I'd say the difference is significant.
```
-rw-r--r-- 1 antti antti 5.2M Apr  1 14:04 app-arm64-v8a-release.apk
-rw-r--r-- 1 antti antti 4.9M Apr  1 14:04 app-armeabi-v7a-release.apk
-rw-r--r-- 1 antti antti 8.4M Apr  1 14:04 app-universal-release.apk
```

PS. I think this issue is a duplicate of #10728 (or at least overlapping)

**Comment by kmdinake** [comment]
@apaatsio I get the same results as you, but when I upload the apks to the playstore for beta testing, only the app-arm64-v8a-release.apk is accepted. Do you perhaps know why that is?

**Comment by rpsnaik** [comment]
Flutter resolved the issue... Switch to master channel and upgrade the packages...
It will download all the dependencies to bind apk and bundles for arm64 and arm devices.
Every thing works same as it said in Documentation <a href="https://flutter.dev/docs/deployment/android">Flutter Release Doc for Android</a>

**Comment by rpsnaik** [comment]
$flutter build apk --release will create a fat apk... 
$flutter build appbundle --release is recomended

**Comment by jonahwilliams** [comment]
Closing, as this is supported by the appbundle or split-per-abi feature. The support for arbitrary renames is tracked elsewhere
